src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -904,7 +903,16 @@ protected QuestionDef parseUpload(IFormElement parent, Element e, int controlUpl
     }
 
     protected QuestionDef parseControl(IFormElement parent, Element e, int controlType) {
-        return parseControl(parent, e, controlType, new Vector<String>());
+        QuestionDef question = parseControl(parent, e, controlType, new Vector<String>());
+
+        if (controlType == Constants.CONTROL_INPUT) {
+            String mediaType = e.getAttributeValue(null, MEDIA_TYPE_ATTR);
+            String appearance = e.getAttributeValue(null, APPEARANCE_ATTR);
+            if ("image/*".equals(mediaType) && MICRO_IMAGE_APPEARANCE_ATTR.equals(appearance)) {
+                question.setControlType(Constants.CONTROL_MICRO_IMAGE);
+            }
+        }

```

Comment:
It feel un-necessary to do any manipulation here with `controlType` but can just decide the widget based on the appearance attribute itself. But if we do want to set a distinct `controlType` for micro image, it's best to do in a new top level method say `parseInput` (similar to `parseUpload` above)